js/webform_civicrm_contact.js

```diff
@@ -9,7 +9,8 @@
         }
         var autocompleteUrl = D.url('webform-civicrm/js/' + field.data('form-id') + '/' + field.data('civicrm-field-key'));
         var isSelect = field.data('is-select');
-        if (!isSelect) {
+        var isAutocomplete = field.data('is-autocomplete');
+        if (isAutocomplete) {
```

Comment:
@jensschuppe I think its just the last argument that needs to be avoided in case of contact id widget. Since the ajax still need to be called for a user that enters contact id in the text field. Can you please confirm if the following update to this PR works for you:

```
diff --git a/js/webform_civicrm_contact.js b/js/webform_civicrm_contact.js
index 5eca49c..7143419 100644
--- a/js/webform_civicrm_contact.js
+++ b/js/webform_civicrm_contact.js
@@ -10,17 +10,20 @@
         var autocompleteUrl = D.url('webform-civicrm/js/' + field.data('form-id') + '/' + field.data('civicrm-field-key'));
         var isSelect = field.data('is-select');
         if (!isSelect) {
+          var tokenValues = field.data('is-contactid') ? false : {
+            hintText: field.data('search-prompt'),
+            noResultsText: field.data('none-prompt'),
+            resultsFormatter: formatChoices,
+            searchingText: "Searching..."
+          };
           wfCivi.existingInit(
             field,
             field.data('civicrm-contact'),
             field.data('form-id'),
             autocompleteUrl,
-            toHide, {
-            hintText: field.data('search-prompt'),
-            noResultsText: field.data('none-prompt'),
-            resultsFormatter: formatChoices,
-            searchingText: "Searching..."
-          });
+            toHide,
+            tokenValues
+          );
         }
 
         field.change(function () {

```

Reply:
Thanks @jitendrapurohit, this seems to work, I added this with 42e1fcc3ef594648252eb75e986deff6f5a1b821.